### DISCUSSION TITLE
Request for GPU Metric Integration Support

### DISCUSSION THREAD

**Discussion by helloxjade** [kedacore/keda#5158] [General]
I would like to use KEDA to perform elastic scaling based on GPU-related metrics. Currently, I'm using Prometheus to collect GPU-related metrics in my Kubernetes cluster, but I'm unsure how to integrate these metrics with KEDA to automatically scale my workloads.

Background:

The workloads I'm running (Deployment/StatefulSet) depend on GPU resources.
I have configured Prometheus to collect GPU-related metrics, such as GPU temperature, utilization, and others.
I have installed and configured KEDA and have successfully used it with other external event triggers.
My Requirements:
I need guidance or suggestions on how to integrate GPU-related metrics collected by Prometheus with KEDA. I want to dynamically scale up or down my workloads based on these GPU metrics.

Specific Questions:

How do I configure the Prometheus Adapter to expose GPU metrics?
How do I create ScaledObject resources in KEDA to respond to GPU metrics?
Are there any examples or documentation available for reference?
Additional Information:

Do I need to create custom metric CRDs?
Examples of Prometheus queries and metric names for GPU metrics.
Please note that providing as much detail and background information as possible will help the community better understand your requirements and provide relevant support and advice. After submitting the issue, you can follow the community's responses and discussions to get further assistance.

**Comment by ioanc** [comment]
Maybe this article might be helpful for your request, and relaying on Prometheus metrics
https://stackoverflow.com/questions/76158750/keda-scaledobject-metric-data-differs-from-prometheus
